act/qc/qcfilter.py

```diff
@@ -1050,6 +1050,43 @@ def datafilter(
             except AttributeError:
                 self._ds[var_name].values = data
 
+            # Adding information on filtering to history attribute
+            flag_masks = None
+            flag_assessments = None
+            flag_meanings = None
+            try:
+                flag_assessments = list(self._ds[qc_var_name].attrs['flag_assessments'])
+                flag_masks = list(self._ds[qc_var_name].attrs['flag_masks'])
+                flag_meanings = list(self._ds[qc_var_name].attrs['flag_meanings'])
+            except KeyError:
+                pass
+
+            # Add comment to history for each test that's filtered out
+            if isinstance(rm_tests, str):
+                rm_tests = [rm_tests]
+            if rm_tests is not None:
+                for test in list(rm_tests):
+                    if test in flag_masks:
+                        index = flag_masks.index(test)
+                        comment = ''.join(['act.qc.datafilter: ', flag_meanings[index]])
+                        if 'history' in self._ds[var_name].attrs.keys():
+                            self._ds[var_name].attrs['history'] += '\n' + comment
+                        else:
+                            self._ds[var_name].attrs['history'] = comment
+
+            if isinstance(rm_assessments, str):
+                rm_assessments = [rm_assessments]
+            if rm_assessments is not None:
+                for assessment in rm_assessments:
+                    if assessment in flag_assessments:
+                        index = [i for i, e in enumerate(flag_assessments) if e == assessment]
+                        for ind in index:
+                            comment = ''.join(['act.qc.datafilter: ', flag_meanings[ind]])
+                            if 'history' in self._ds[var_name].attrs.keys():
+                                self._ds[var_name].attrs['history'] += '\n' + comment
+                            else:
```

Comment:
More for clarification if rm_tests exist and appends to the comment, will rm_assessments then append to that as well? I think I understand what's happening here

Reply:
Yep, essentially comment is just a dummy variable that is being appended to the history attribute for both rm_assessments and rm_tests.  It's a little more complicated for rm_assessments though since it could just be "Bad" and we need to find all the bad tests.